### PR TITLE
MAINT: Remove `distutils` usage

### PR DESCRIPTION
**Pull request by Smit-create**
Fixing a few `distutils` warnings like:
```py
/scipy/setup.py:123: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.command.sdist import sdist
```

cc @rgommers

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def __exit__(self, exception_type, exception_value, traceback):
             f.write(self.bsd_text)
 
 
-from distutils.command.sdist import sdist
+from setuptools.command.sdist import sdist
```
Comment (rgommers) [inline setup.py:123]: We can't do this, this is a backwards incompatible change.

**Review by rgommers — COMMENTED** [review]
Thanks @Smit-create. Overall this looks good, avoiding `distutils` in code generation scripts is useful long term. 

Let's not try to change `setup.py` files - we're going to get rid of them anyway, and also this can lead to subtle changes in behavior.

**Comment by rgommers** [comment]
Ah, CI fails because `_lib` isn't the right place. If you put it in `_build_utils`, things may work better.

**Comment by Smit-create** [comment]
> If you put it in `_build_utils`, things may work better.

That should probably fail on `meson` because we don't add `_build_utils` there.

**Comment by rgommers** [comment]
Ah that is something to look into - we use at least `tempita.py` from there, so we need to add `_build_utils` in the sdist. But indeed we don't need anything from there at runtime, so we don't need to install `_build_utils`. I think it'll work as is, if not you may have to do the same `importlib` trick as in `dev.py`.

**Comment by Smit-create** [comment]
`meson` tests work fine and so does locally for me. Some path issues while building on CI.

**Comment by rgommers** [comment]
@Smit-create how about we make things easy and just copy `def newer: ...` into each of the four files that need it? That `newer` code is a lot more readable than the `importlib` trick, and not many more lines of code.

**Comment by Smit-create** [comment]
I think after these removals of `distutils`, we only have two usages left:
1. `get_python_inc` - All of them are in `*/setup.py`, so not a big concern
2. `get_platform` - In `_build_utils/_fortran.py`

**Review by rgommers — APPROVED** [review]
Thanks @Smit-create, this LGTM now.

**Comment by rgommers** [comment]
> 2. `get_platform` - In `_build_utils/_fortran.py`

this one may need a careful look. it seems that it is not identical to `sysconfig.get_platform`.

**Pull request merged**